pkg/collector/corechecks/checkbase.go

```diff
@@ -295,3 +298,8 @@ func (c *CheckBase) GetDiagnoses() ([]diagnosis.Diagnosis, error) {
 func (c *CheckBase) IsHAEnabled() bool {
 	return c.haEnabled
 }
+
+// IsHASupported returns if the check is compatible with High Availability
+func (c *CheckBase) IsHASupported() bool {
+	return false
```

Comment:
How checks would opt-in? Since go doesn't support virtual methods, `CommonConfigure` will always call this exact implementation, and will always get false.